source/audio/Audio.cpp

```diff
@@ -431,9 +453,33 @@ void Audio::Step()
 		// Check if the source has stopped (i.e. because it ran out of buffers).
 		ALint state;
 		alGetSourcei(musicSource, AL_SOURCE_STATE, &state);
-		if(state != AL_PLAYING)
+		if(state != AL_PLAYING && state == AL_PAUSED)
 			alSourcePlay(musicSource);
 	}
+
+	if(shouldPause && !shouldResume)
+	{
+		ALint state;
+		for(const Source &source : sources)
+		{
+			alGetSourcei(source.ID(), AL_SOURCE_STATE, &state);
+			if(state == AL_PLAYING)
+				alSourcePause(source.ID());
+		}
+		shouldPause = false;
+	}
+
+	if(shouldResume)
+	{
+		ALint state;
+		for(const Source &source : sources)
+		{
+			alGetSourcei(source.ID(), AL_SOURCE_STATE, &state);
+			if(state == AL_PAUSED)
+				alSourcePlay(source.ID());
+		}
+		shouldResume = false;
+	}
```

Comment:
restructure the `if` blocks
```suggestion
	if(shouldResume)
	{
		ALint state;
		for(const Source &source : sources)
		{
			alGetSourcei(source.ID(), AL_SOURCE_STATE, &state);
			if(state == AL_PAUSED)
				alSourcePlay(source.ID());
		}
		shouldResume = false;
	}
	else if(shouldPause)
	{
		ALint state;
		for(const Source &source : sources)
		{
			alGetSourcei(source.ID(), AL_SOURCE_STATE, &state);
			if(state == AL_PLAYING)
				alSourcePause(source.ID());
		}
		shouldPause = false;
	}
```

Reply:
I think this doesn't reset the shouldPause flag if they are both set? That's a simple fix, though.